cr17/nodes/cr17/high_level_state_controller.py

```diff
@@ -40,32 +34,65 @@ def __init__(self):
         self.state_pub = rospy.Publisher(STATE_TOPIC, String, queue_size=10)
 
         self.pose = Pose()
-        self.pathPoses = []     #TODO: maybe remove, send entire current path to navigator
-        self.curPathIndex = 0   #TODO: maybe remove, send entire current path to navigator
+        rospy.Subscriber(CUR_ARM_ANGLE_TOPIC, Float64, self.arm_sub)
+        self.arm_cur_angle = 45.0
+        rospy.Subscriber(PATH_TOPIC, Path, self.path_sub)
+        self.pathPoses = []
+        self.curPathIndex = 0
+        rospy.Subscriber(JOY_TOPIC, Joy, self.joy_sub)
         self.autostate = "INIT"
+        self.highState = "AUTO"
+        self.lastJoy = None
+        self.leftDrive = None
+        self.rightDrive = None
+        self.armDown = None
+        self.armUp = None
+        self.handDown = None
+        self.handUp = None
+        self.teleopButton_prev = 0
```

Comment:
Yea, this is not the correct high_level_state_controller, this merge will wipe lots of changes to it